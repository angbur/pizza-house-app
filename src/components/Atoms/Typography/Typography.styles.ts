import styled from 'styled-components';

export const PageTitle = styled.h2`
  font-size: 'Source Sans Pro Light';
  font-weight: 300;
  line-height: 1.3;
  letter-spacing: 1px;
  font-size: 2.25rem;
`;
