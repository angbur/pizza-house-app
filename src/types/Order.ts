import type { Pizza } from 'types/Pizza';

export type OrderItem = {
  _id: string;
  item: Omit<Pizza, 'categories' | '_id'>;
  quantity: number;
};
