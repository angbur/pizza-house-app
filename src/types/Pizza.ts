export const ingredients = [
  'tomato sauce',
  'pineapple',
  'white sauce',
  'cheese',
  'mozzarella',
  'tomatoes',
  'mushrooms',
  'chili',
  'olives',
  'pepper',
  'corn',
  'onion',
  'salami',
  'bacon',
  'shrimps',
  'red fish',
  'basil',
  'arugula',
  'oregano',
  'parsley',
] as const;
export const categories = ['spicy', 'vegetarian', 'meaty', 'seafood'] as const;

export type Ingredients = typeof ingredients[number];
export type Categories = typeof categories[number];

export type Pizza = {
  _id: string;
  name: string;
  ingredients: Ingredients[];
  categories: Categories[];
  price: number;
};
