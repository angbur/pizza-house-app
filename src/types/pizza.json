{
    "name": "Tipsy Tomato",
    "ingredients":["tomato sauce", "tomatoes", "olives", "mozzarella", "cheese", "parsley", "basil"],
    "price": 20,
    "categories": ["vegetarian"]
},
{
    "name": "Pinza Mare",
    "ingredients": ["shrimps", "red fish", "olives", "white sauce", "oregano", "basil"],
    "price": 26,
    "categories": ["seafood"]
},
{
    "name": "The Crispy Crust",
    "ingredients": ["tomato sauce", "salami", "oregano", "olives", "cheese", "basil"],
    "price": 30,
    "categories": ["meaty"]
},
{
    "name": "Feeling Saucy",
    "ingredients": ["tomato sauce", "pineapple", "corn", "olives", "onion", "arugula"],
    "price": 28,
    "categories": ["vegetarian"]
},
{
    "name": "Mellow",
    "ingredients": ["mushrooms", "onion", "basil", "parsley", "tomato sauce"],
    "price": 24,
    "categories": ["vegetarian"]
},
{
    "name": "Heirloom",
    "ingredients": ["mushrooms", "onion", "basil", "parsley", "white sauce", "olives"],
    "price": 24,
    "categories": ["vegetarian"]
},
{
    "name": "Lighthouse",
    "ingredients": ["white sauce", "cheese", "pepper", "chili", "parsley", "basil"],
    "price": 24,
    "categories": ["vegetarian", "spicy"]
},
{
    "name": "Lighthouse",
    "ingredients": ["tomato sauce", "bacon", "olives", "cheese", "parsley", "basil"],
    "price": 24,
    "categories": ["meaty"]
}