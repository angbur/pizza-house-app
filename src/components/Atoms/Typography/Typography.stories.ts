import { Meta, StoryObj } from '@storybook/react';
import { themeProperties } from 'components/Theme/theme.types';
import Typography from './Typography';

const meta = {
  title: 'Atoms/Typography',
  component: Typography,
  tags: ['docsPage'],
  argTypes: {
    variant: themeProperties.typographyNames,
    color: themeProperties.colorNames,
  },
} satisfies Meta<typeof Typography>;

export default meta;

type Story = StoryObj<typeof meta>;

export const MainTitle: Story = {
  args: {
    variant: 'main-title',
    color: 'primary',
    children: 'Hello in my world',
  },
};
