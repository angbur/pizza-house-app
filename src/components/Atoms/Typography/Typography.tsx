import clsx from 'clsx';
import { PropsWithChildren } from 'react';
import { CSSProperties } from 'styled-components';
import { ColorVariants } from '../../Theme/theme.types';

const variants = ['mainTitle', 'sectionTitle','cardTitle', 'itemTitle', 'paragraph', 'caption'] as const;
type TypographyVariant = typeof variants[number];

const variantToElement = {
  'mainTitle': 'h1',
  'sectionTitle': 'h2',
  'cardTitle': 'h3',
  'itemTitle': 'h5',
  'paragraph': 'p',
  'caption': 'p',
} as const;

export type TypographyProps = {
  variant?: TypographyVariant;
  color: ColorVariants;
  className?: string;
  style?: CSSProperties;
};

const Typography = ({
    variant,
    className,
    style,
    color,
    children
}: PropsWithChildren<TypographyProps>) => {
  const Element = variantToElement[variant ? variant : 'paragraph'];

  return <Element className={clsx(variant, className)} style={theme.typography[variant]}>{children}</Element>
};

export default Typography;
