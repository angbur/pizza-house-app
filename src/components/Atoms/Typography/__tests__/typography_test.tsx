import renderer from 'react-test-renderer';
import ThemeProvider from 'components/Theme/ThemeContext';
import Typography from '../Typography';
import { TypographyVariant } from 'components/Theme/theme.types';

type WrapperProps = {
  variant: TypographyVariant;
};

const Wrapper = ({ variant }: WrapperProps) => {
  return (
    <ThemeProvider>
      <Typography variant={variant}>Text</Typography>
    </ThemeProvider>
  );
};

describe('Typography', () => {
  it('renders correctly main-title', () => {
    const tree = renderer.create(<Wrapper variant={'main-title'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly section-title', () => {
    const tree = renderer.create(<Wrapper variant={'section-title'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly card-title', () => {
    const tree = renderer.create(<Wrapper variant={'card-title'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly card-subtitle', () => {
    const tree = renderer.create(<Wrapper variant={'card-subtitle'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly dialog-title', () => {
    const tree = renderer.create(<Wrapper variant={'dialog-title'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly item-title', () => {
    const tree = renderer.create(<Wrapper variant={'item-title'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly paragraph', () => {
    const tree = renderer.create(<Wrapper variant={'paragraph'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly price-tag', () => {
    const tree = renderer.create(<Wrapper variant={'price-tag'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly caption', () => {
    const tree = renderer.create(<Wrapper variant={'caption'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
