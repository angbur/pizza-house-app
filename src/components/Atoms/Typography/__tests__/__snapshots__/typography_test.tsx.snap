// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography renders correctly caption 1`] = `
<p
  className="caption"
  style={
    Object {
      "color": "none",
      "fontFamily": "Klepon Ijo",
      "fontSize": "2rem",
      "fontWeight": "400",
      "letterSpacing": "0",
      "lineHeight": "1.6",
    }
  }
>
  Text
</p>
`;

exports[`Typography renders correctly card-subtitle 1`] = `
<h4
  className="card-subtitle"
  style={
    Object {
      "color": "none",
      "fontFamily": "Source Sans Pro",
      "fontSize": "0.9rem",
      "fontWeight": "400",
      "lineHeight": "1.6",
    }
  }
>
  Text
</h4>
`;

exports[`Typography renders correctly card-title 1`] = `
<h3
  className="card-title"
  style={
    Object {
      "color": "none",
      "fontFamily": "Klepon Ijo",
      "fontSize": "2rem",
      "fontWeight": "400",
      "letterSpacing": "2px",
      "lineHeight": "1.2",
    }
  }
>
  Text
</h3>
`;

exports[`Typography renders correctly dialog-title 1`] = `
<h2
  className="dialog-title"
  style={
    Object {
      "color": "none",
      "fontFamily": "Source Sans Pro",
      "fontSize": "1.6rem",
      "fontWeight": "600",
      "letterSpacing": "1px",
      "lineHeight": "1.6",
    }
  }
>
  Text
</h2>
`;

exports[`Typography renders correctly item-title 1`] = `
<h5
  className="item-title"
  style={
    Object {
      "color": "none",
      "fontFamily": "Source Sans Pro",
      "fontSize": "1rem",
      "fontWeight": "600",
      "letterSpacing": "1px",
      "lineHeight": "1.4",
    }
  }
>
  Text
</h5>
`;

exports[`Typography renders correctly main-title 1`] = `
<h1
  className="main-title"
  style={
    Object {
      "color": "none",
      "fontFamily": "Klepon Ijo",
      "fontSize": "3rem",
      "fontWeight": "400",
      "letterSpacing": "1px",
      "lineHeight": "1.6",
    }
  }
>
  Text
</h1>
`;

exports[`Typography renders correctly paragraph 1`] = `
<p
  className="paragraph"
  style={
    Object {
      "color": "none",
      "fontFamily": "Source Sans Pro",
      "fontSize": "1rem",
      "fontWeight": "400",
      "letterSpacing": "0",
      "lineHeight": "1.6",
    }
  }
>
  Text
</p>
`;

exports[`Typography renders correctly price-tag 1`] = `
<p
  className="price-tag"
  style={
    Object {
      "color": "none",
      "fontFamily": "Klepon Ijo",
      "fontSize": "3rem",
      "letterSpacing": "0",
    }
  }
>
  Text
</p>
`;

exports[`Typography renders correctly section-title 1`] = `
<h2
  className="section-title"
  style={
    Object {
      "color": "none",
      "fontFamily": "Source Sans Pro",
      "fontSize": "2.25rem",
      "fontWeight": "300",
      "letterSpacing": "1px",
      "lineHeight": "1.3",
    }
  }
>
  Text
</h2>
`;
